### PR TITLE
controller: echo ip address if destination service receives ip

### LINKED ISSUE #181: Allow sending an IP address to the Destination service
In some cases (TCP, HTTP/1 without a Host), we may not know a DNS name to lookup in the Destination service, other than the original dst socket address.

As a first step, we should accept IP addresses, and just echo back the same address. In the future, we can consider doing something fancier.

See #167 for some more context.

### PR DESCRIPTION
**Pull request by seanmonstar**
Closes #181

### CODE REVIEW

File: controller/destination/server.go [hunk 1]
```diff
@@ -126,6 +132,29 @@ func (s *server) Get(dest *common.Destination, stream pb.Destination_GetServer)
 	return err
 }
 
+func (s *server) echoIPDestination(host string, port int, stream pb.Destination_GetServer) error {
+	ip, err := util.ParseIPV4(host)
+	if err != nil {
+		return err
+	}
+	addrs := make([]common.TcpAddress, 1)
```
Comment (siggy) [inline controller/destination/server.go]: consider stack allocation when you know the size ahead of time:
```go
addrs := []common.TcpAddress{
	{
		Ip:   ip,
		Port: uint32(port),
	},
}
```

from https://segment.com/blog/allocation-efficiency-in-high-performance-go-services/:
```
Go prefers allocation on the stack
```

File: controller/destination/server.go [hunk 2]
```diff
@@ -126,6 +132,29 @@ func (s *server) Get(dest *common.Destination, stream pb.Destination_GetServer)
 	return err
 }
 
+func (s *server) echoIPDestination(host string, port int, stream pb.Destination_GetServer) error {
```
Comment (siggy) [inline controller/destination/server.go]: as you don't need the server here, you can just declare the function standalone:
```golang
func echoIPDestination(host...
```

**Comment by seanmonstar** [comment]
> ```go
> isIp, err := isIPAddress(host)
> if err != nil {
>   return err
> }
> ```

But we are parsing in a way that very well may error, if it is a DNS name, and if so, we just wan to continue along...

**Comment by adleong** [comment]
My instinct would be to capture the parse error inside the implementation of `isIPAddress` and return false.  `isIPAddress` may not even need to have an error in its return type at all.

**Comment by seanmonstar** [comment]
OK, so I think I'm gather that the issue is returning an `error` instead of something else. I can fix that up.

**Comment by adleong** [comment]
Or even something like:

```
isIP, IP, port := isIPAddress(host)
if isIP {
  return s.echoIPDestination(IP, port)
}
```

**Comment by seanmonstar** [comment]
OK, updating to an `if isIP {`...

**Pull request merged**